apps/api/src/domain/models/audio-visual/audio-item/repositories/arango-audio-item-query-repository.spec.ts

```diff
@@ -422,6 +423,75 @@ describe(`ArangoAudioItemQueryRepository`, () => {
         });
     });
 
+    describe(`connectResourcesWith`, () => {
+        const targetAudioItem = buildTestInstance(EventSourcedAudioItemViewModel, {
+            connections: [],
+        });
+
+        beforeEach(async () => {
+            await databaseProvider.clearViews();
+
+            await testQueryRepository.create(targetAudioItem);
+        });
+
+        it(`should add the connection info`, async () => {
+            const generalContext: IEdgeConnectionContext = {
+                type: EdgeConnectionContextType.general,
+            };
+
+            const otherCompositeIdentifier = {
+                type: 'widget' as ResourceType,
+                id: buildDummyUuid(67),
+            };
+
+            const noteId = buildDummyUuid(76);
+
+            const textForNote = 'This is why the widget is relevant to the audio item';
+
+            const langaugeCodeForNote = LanguageCode.Chilcotin;
+
+            const role = EdgeConnectionMemberRole.to;
+
+            await testQueryRepository.connectResourcesWith(targetAudioItem.id, {
+                noteId,
+                selfContext: generalContext,
+                otherContext: generalContext,
+                compositeIdentifier: otherCompositeIdentifier,
+                text: buildMultilingualTextWithSingleItem(textForNote, langaugeCodeForNote),
+                role,
+            });
+
+            const { connections } = (await testQueryRepository.fetchById(
+                targetAudioItem.id
+            )) as EventSourcedAudioItemViewModel;
+
+            expect(connections).toHaveLength(1);
+
+            const {
+                selfContext,
+                otherCompositeIdentifier: foundCompositeIdentifierForConnectedResource,
+                otherContext,
+                note,
+                role: edgeConnectionMemberRole,
+            } = connections[0];
+
+            expect(selfContext).toEqual(generalContext);
```

Comment:
I think it's fine not to check the other context types. The general one is the only one we can count on being applicable for the other models. It would be very strange to some how drop the additional props of, say, a "time range" context in our AQL implementation, and we will surely have tons of higher level tests anyway.